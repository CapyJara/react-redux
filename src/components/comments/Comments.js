import React from 'react';
import PropTypes from 'prop-types';
import Comment from './Comment';

function Comments({ comments }) {
  const commentList = comments.map((comment, i) => (

    <li key={`${comment.title}-${i}`}>
      <Comment comment={comment} />
    </li>
  ));

  return (
    <ul>
      {commentList}
    </ul>
  );
}

Comments.propTypes = {
  comments: PropTypes.array.isRequired
};

export default Comments;
