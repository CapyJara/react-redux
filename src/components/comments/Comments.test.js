import React from 'react';
import { shallow } from 'enzyme';
import Comments from './Comments';

describe('Comments component', () => {
  it('renders a Comments', () => {

    const comments = [
      {
        title: 'title of my comment',
        comment: 'this is my comment'
      },
      {
        title: 'by me',
        comment: 'this is me comment'
      },
      {
        title: 'by you',
        comment: 'this you comment'
      },
    ];

    const wrapper = shallow(<Comments comments={comments}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
