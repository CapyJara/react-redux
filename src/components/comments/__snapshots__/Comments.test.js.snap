// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component renders a Comments 1`] = `
<ul>
  <li
    key="title of my comment-0"
  >
    <Comment
      comment={
        Object {
          "comment": "this is my comment",
          "title": "title of my comment",
        }
      }
    />
  </li>
  <li
    key="by me-1"
  >
    <Comment
      comment={
        Object {
          "comment": "this is me comment",
          "title": "by me",
        }
      }
    />
  </li>
  <li
    key="by you-2"
  >
    <Comment
      comment={
        Object {
          "comment": "this you comment",
          "title": "by you",
        }
      }
    />
  </li>
</ul>
`;
